File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-0.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        
        answer:int = 0
        p1 = 0
        p2 = 0

        while p1 < len(s):
            temp_str = ""

            for c in s[p1:]:
                if c in temp_str:
                    answer = max(len(temp_str), answer)
                    break
                
                temp_str += c
            answer = max(len(temp_str), answer)
            p1 += 1
        
        return answer

        

File: Data Structures & Algorithms/two-integer-sum/submission-0.java
class Solution {
    public int[] twoSum(int[] nums, int target) {

// Create answer array
// Create HashMap
// Iterate through the entire nums array
// Create one variables | difference
// If the difference is within the 

        int[] answer_array = {0,0};
        HashMap<Integer, Integer> map = new HashMap<>();

        for(int i = 0; i < nums.length; i++)
        {
            int num = nums[i];
            int diff = target - num;

            if(map.containsKey(diff))
            {
                answer_array[0] = map.get(diff);
                answer_array[1] = i;
                break;
            }

            map.put(num, i);
        }

        return answer_array;
    }
}
